core/src/main/java/org/apache/iceberg/rest/InMemoryPlanningState.java

```diff
@@ -68,6 +73,37 @@ void addNextPlanTask(String currentTask, String nextTask) {
     planTaskToNext.put(currentTask, nextTask);
   }
 
+  void addAsyncPlan(String plan) {
+    ongoingAsyncPlans.add(plan);
+  }
+
+  PlanStatus asyncPlanStatus(String plan) {
+    if (failedAsyncPlans.contains(plan)) {
+      return PlanStatus.FAILED;
+    }
+
+    if (ongoingAsyncPlans.contains(plan)) {
+      return PlanStatus.SUBMITTED;
+    }
```

Comment:
Lets say if I call CANCEL Api it may happen that we will still show the client the SUBMITTED status